core/src/main/scala/org/apache/spark/h2o/backends/internal/H2ORpcEndpoint.scala

```diff
@@ -48,12 +49,19 @@ class H2ORpcEndpoint(override val rpcEnv: RpcEnv) extends ThreadSafeRpcEndpoint
 
     case StopEndpointMsg =>
       this.stop()
+    case LockClusterMsg() =>
+      Paxos.lockCloud("Locking the cloud from Sparkling Water as we have reached the expected cluster size.")
   }
 
   override def receiveAndReply(context: RpcCallContext): PartialFunction[Any, Unit] = {
     case StartH2OWorkersMsg(conf) =>
       val nodeDesc = InternalH2OBackend.startH2OWorker(conf)
       context.reply(nodeDesc)
+    case CheckClusterSizeMsg() =>
+      context.reply(H2O.CLOUD.size())
+    case IsLeaderNodeMsg() =>
+      val currentNode = NodeDesc(SparkEnv.get.executorId, H2O.SELF_ADDRESS.getHostAddress, H2O.API_PORT)
```

Comment:
is `H2O.SELF_ADDRESS.getHostAddress` going to be an IP address in 100% cases?

Reply:
Yup, from doc
>      * Returns the IP address string in the textual presentation.

But in the PR  I'm working on (with the fix for HTTPS) I will change this anyways to hostname so both IP and hostname is just fine here. I just did not want to this as part of this PR so the following PR contains the whole change.

Reply:
But just to make it clearer for the future, I have changed the method to send only NodeDesc of the leader node so worker nodes are not exposed at all